### PR TITLE
Update composer for CMS 5 compatibility

### PR DESCRIPTION
**Pull request by kevingroeger**
-> Tested on a large-scale project

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -16,9 +16,9 @@
         }
     ],
     "require": {
-        "silverstripe/framework": "^4@dev",
+        "silverstripe/framework": "^5",
```
Comment (michalkleiner) [inline composer.json:19]:
```suggestion
        "silverstripe/framework": "^4 || ^5",
```

File: composer.json [hunk 2]
```diff
@@ -16,9 +16,9 @@
...
         "league/flysystem-aws-s3-v3": "^1.0",
-        "silverstripe/vendor-plugin": "^1.0",
+        "silverstripe/vendor-plugin": "^2",
```
Comment (michalkleiner) [inline composer.json:21]:
```suggestion
        "silverstripe/vendor-plugin": "^1 || ^2",
```

**Comment by michalkleiner** [comment]
Thanks for the PR @kevingroeger. If the composer requirements are the only change needed, we can keep both 4 and 5 as well as 1 and 2 for the vendor-plugin to keep supporting both 4 and 5 versions of the CMS, right?

**Comment by kevingroeger** [comment]
Thanks for the quick reply @michalkleiner

I applied your suggested changes

**Comment by michalkleiner** [comment]
Seems like not both of them?

**Comment by kevingroeger** [comment]
Yes, sorry. Just realised that when I tested it on SilverStripe4 again to be safe. 

I applied the other one as well now. 

Everything worked fine in both versions. Tested on 4.13 and 5.0

**Comment by michalkleiner** [comment]
Thanks for that. No deprecations, no warnings in dev mode etc?

**Comment by kevingroeger** [comment]
Surprisingly not. I would've expected some, but I haven't gotten any errors or warnings (in dev mode). Even searched my logs and didn't find any issues whatsoever.

**Pull request merged**